### PR TITLE
Immediate scheduler

### PR DESCRIPTION
**Pull request by arturscussel**
It's already without the defines. Thanks

### CODE REVIEW

**Comment by rhornig** [comment]
Hello @tuexen @ruengeler ,
we have received a pull request that turns on the immediate mode for pcap. As the documentations suggest, the scheduler was not running in immediate mode until now. According to the docs http://www.manpages.info/macosx/pcap.3.html the read operation waits for several packets to arrive until the timeout expires so we are not seeing the packets as they arrive.

Would you confirm that it's OK to use the immediate mode? It seems justified IMHO for something that's called "realtime", but you may have some other experiences. Please give an OK, if we can merge.

File: src/linklayer/ext/cSocketRTScheduler.cc [hunk 1]
```diff
@@ -117,11 +117,18 @@ void cSocketRTScheduler::setInterfaceModule(cModule *mod, const char *dev, const
 
     /* get pcap handle */
     memset(&errbuf, 0, sizeof(errbuf));
-    if ((pd = pcap_open_live(dev, PCAP_SNAPLEN, 0, PCAP_TIMEOUT, errbuf)) == NULL)
-        throw cRuntimeError("cSocketRTScheduler::setInterfaceModule(): Cannot open pcap device, error = %s", errbuf);
+    if((pd = pcap_create(dev,errbuf)) == NULL)
```
Comment (tuexen) [inline src/linklayer/ext/cSocketRTScheduler.cc]: Use 
pcap_create(dev, errbuf)

File: src/linklayer/ext/cSocketRTScheduler.cc [hunk 2]
```diff
@@ -117,11 +117,18 @@ void cSocketRTScheduler::setInterfaceModule(cModule *mod, const char *dev, const
...
+        throw cRuntimeError("cSocketRTScheduler::setInterfaceModule(): Cannot create pcap device, error = %s", errbuf);
     else if (strlen(errbuf) > 0)
         EV << "cSocketRTScheduler::setInterfaceModule(): pcap_open_live returned warning: " << errbuf << "\n";
 
+    /* apply the immediate mode to pcap */
+    if(pcap_set_immediate_mode(pd,1) != 0)
```
Comment (tuexen) [inline src/linklayer/ext/cSocketRTScheduler.cc]: Use
pcap_set_immediate_mode(pd, 1)

File: src/linklayer/ext/cSocketRTScheduler.cc [hunk 3]
```diff
@@ -220,7 +227,9 @@ bool cSocketRTScheduler::receiveWithTimeout()
 
     found = false;
     timeout.tv_sec = 0;
-    timeout.tv_usec = PCAP_TIMEOUT * 1000;
+    timeout.tv_usec = 100;
```
Comment (tuexen) [inline src/linklayer/ext/cSocketRTScheduler.cc]: Maybe it is better to use some #define the the timeout. Then is it easier to change parameters, if needed.

**Comment by tuexen** [comment]
The changes with my editorial changes look good.
However, I'm not sure if the functions newly called are available in the pcap implementations used on the platforms supported by inet. If that is true, the patch should go in.

**Comment by rhornig** [comment]
Thanks, I will check whether the pull request builds correctly on all platforms.

**Comment by avarga** [comment]
Found an issue in cSocketRTScheduler, see #120

**Comment by rhornig** [comment]
Reformatted a bit and made the timeout is now defined with a macro. Merged into master as 81596a.

**Pull request closed without merging**